spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/ConfigServiceTemplate.java

```diff
@@ -126,6 +124,16 @@ public ConfigServiceTemplate(ConfigHttpClient httpClient, ConnectionStringPool c
         return result;
     }
 
+    private boolean itemsReachedRange(List<KeyValueItem> items, Range range) {
+        if (range == null) {
+            return false;
+        }
+
+        int maxSize = range.getEndItem() - range.getStartItem() + 1;
```

Comment:
is there any possibility that `endItem() < startItem()` ?

Reply:
No, Range is not allowed to be created with `startItem <= endItem`.